tests/behat/filter.feature

```diff
@@ -0,0 +1,76 @@
+      | activity    | name               | intro                     | course | idnumber     | publishnewquestion |
+      | studentquiz | StudentQuiz Test 1 | StudentQuiz description 1 | C1     | studentquiz1 | 1                  |
+    And the following "questions" exist:
+      | questioncategory               | qtype     | name            | questiontext          |
+      | Default for StudentQuiz Test 1 | truefalse | Test question 1 | Answer the question 1 |
+      | Default for StudentQuiz Test 1 | truefalse | Test question 2 | Answer the question 2 |
+
+  @javascript @_switch_window
+  Scenario: Check validation client in the advance filter.
+    When I am on the "StudentQuiz Test 1" "mod_studentquiz > View" page logged in as "admin"
+    And I should see "Show more..."
+    Then I click on "Show more..." "link"
+    And I set the field "Rating value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "Rating value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "Rating value" to ""
+    And I should not see "You must enter a number that is greater than 0."
+
+    And I set the field "Difficulty value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "Difficulty value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "Difficulty value" to ""
+    And I should not see "You must enter a number that is greater than 0."
+
+    And I set the field "Comments value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "Comments value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "Comments value" to ""
+    And I should not see "You must enter a number that is greater than 0."
+
+    And I set the field "My attempts value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "My attempts value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "My attempts value" to ""
+    And I should not see "You must enter a number that is greater than 0."
+
+    And I set the field "My difficulty value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "My difficulty value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "My difficulty value" to ""
+    And I should not see "You must enter a number that is greater than 0."
+
+    And I set the field "My Rating value" to "TF 01"
+    And I should see "You must enter a number here."
+    And I set the field "My Rating value" to "-2"
+    And I should see "You must enter a number that is greater than 0."
+    And I set the field "My Rating value" to ""
+    And I should not see "You must enter a number that is greater than 0."
```

Comment:
I have to say, if it was me, I would find a way to check all this detailed validation logic in PHPunit (faster!) and then all the Behat test has to verify is that one typical validation message appears in the UI. However, you have coded this now, so you could leave it as is.